xbmc/music/MusicDatabase.cpp

```diff
@@ -502,7 +502,18 @@ bool CMusicDatabase::AddAlbum(CAlbum& album, int idSource)
       PrepareSQL("UPDATE album SET iDisctotal = (SELECT COUNT(DISTINCT iTrack >> 16) FROM song "
                  "WHERE song.idAlbum = album.idAlbum) WHERE idAlbum = %i",
                  album.idAlbum));
-
+  if (!album.bBoxedSet && !album.bCompilation)
+  {
+    std::string strSQL;
+    strSQL = PrepareSQL("SELECT COUNT(DISTINCT strDiscSubtitle) FROM song WHERE song.idAlbum = %i",
+                        album.idAlbum);
+    int numTitles = static_cast<int>(strtol(GetSingleValue(strSQL).c_str(), nullptr, 10));
+    if (numTitles >=3) // 3 or more distinct disc titles so this is a boxset album
+    {
+    strSQL = PrepareSQL("UPDATE album SET bBoxedSet=1 WHERE album.idAlbum=%i", album.idAlbum);
+    m_pDS->exec(strSQL);
```

Comment:
Indentation (save the "bot" on master)

And I would still rather see the comment labelling this entire section of code than inline after the 2nd if

Reply:
Missed the indentation, thanks!

Regarding the comment, is this better above the if (!album.bBoxedSet && .....)
`  /* If an album is not already marked as a boxset or compilation, check how many distinct
     disc titles we already have.  If there are at least three, then the album is a boxset so mark
     it as such in the db. This allows for boxsets spread over multiple directories (CD1, CD2 etc)
  */`
The closing comment tag `*/` is not on the same line as it is above, it's on its own line.

Reply:
Humm... too verbose, find a happy medium (God I'm in a picky mode tonight, fetch me a beer!).

 // Set a non-compilation album with at least 3 distinct disc titles as a boxset

Reply:
Grrr, who would have thought comments would be so difficult to get right :)

Reply:
Silly isn't it, but the quality work deserves it :)